libflux/src/semantic/parser/grammar.md

```diff
@@ -0,0 +1,47 @@
+# Polytype Parser Grammar 
+
+[Modified from Joshua Lapacik's work here](https://github.com/jlapacik/ftel/blob/master/antlr/flux.g4)
+
+```
+'?' = optional; once or none 
+'*' = zero or more times
+'|' = or 
+(values) = a list of possibly repeating values
+single quotes indicate a literal value
+
+
+polytype    = 'forall' '[' vars? ']' ('where' constraints)? monotype
+
+vars        = typeVar (',' typeVar)* 
+constraints = constraint ( (',' | 'and') constraint)* 
+constraint  = typeVar ('is' | ':') kinds
+kinds       = kind ( '+' kind)*
+kind        = IDENTIFIER 
+monotype    = typeVar | typ | array | row | function
+
+type_var    = IDENTIFIER 
```

Comment:
We made a change to this so that all strings which aren't keywords are identifiers. I forgot to change it in the grammar. I'll go ahead and push that change.